File: Source/CSharp/Expressive/Expressive/Functions/CeilingFunction.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Expressive.Functions
{
    internal class CeilingFunction : FunctionBase
    {
        #region FunctionBase Members

        public override string Name { get { return "Ceiling"; } }

        public override object Evaluate(object[] values)
        {
            this.ValidateParameterCount(values, 1, 1);

            if (values[0] is double)
            {
                return Math.Ceiling((double)values[0]);
            }
            else if (values[0] is decimal)
            {
                return Math.Ceiling((decimal)values[0]);
            }
            return Math.Ceiling(Convert.ToDouble(values[0]));
        }

        #endregion
    }
}

File: Source/Java/Expressive/src/main/java/com/bijington/expressive/functions/mathematical/CeilingFunction.java
package com.bijington.expressive.functions.mathematical;

import com.bijington.expressive.exceptions.ExpressiveException;
import com.bijington.expressive.expressions.IExpression;
import com.bijington.expressive.functions.FunctionBase;
import com.bijington.expressive.helpers.Convert;

/**
 * Created by shaun on 18/07/2016.
 */
public class CeilingFunction extends FunctionBase {
    @Override
    public String getName() {
        return "Ceiling";
    }

    @Override
    public Object evaluate(IExpression[] parameters) throws ExpressiveException {
        this.validateParameterCount(parameters, 1, 1);

        Double value = Convert.toDouble(parameters[0].evaluate(this.getVariables()));

        return Math.ceil(value);
    }
}
